ariadne/asgi.py

```diff
@@ -253,5 +253,8 @@ def __init__(
                 {"type": GQL_DATA, "id": operation_id, "payload": payload}
             )
 
-        if websocket.application_state != WebSocketState.DISCONNECTED:
+        if (
```

Comment:
Do we need to check both conditions or does client disconnecting imply application state being disconnected as well?

Reply:
I think we need check both.

When a client sends `GQL_CONNECTION_TERMINATE`, server's `handle_websocket_message(...)` calls `websocket.close()` which immediately change the application_state DISCONNECTED (but the client_state is still CONNECTED)